### PR TITLE
update sample json format

### PR DESCRIPTION
**Pull request by pq**
Updates to latest index format.

* adds `sourceLocation`
* updates display label

image

Follow-up to add a popup with description details tracked in https://github.com/flutter/flutter-intellij/issues/3088.

/cc @jacob314 @devoncarew

### CODE REVIEW

File: src/io/flutter/samples/FlutterSample.java [hunk 1]
```diff
@@ -17,22 +17,31 @@
...
   public String toString() {
-    return getElement() + " (" + getLibrary() + ")";
+    return getDisplayLabel();
+  }
+
+  /**
+   * Get a label suitable for display in a chooser (e.g., combobox).
+   */
+  public String getDisplayLabel() {
+    // TODO(pq): add disambiguation once it's needed.
```
Comment (jacob314) [inline src/io/flutter/samples/FlutterSample.java]: also looks like the existing labels aren't that user friendly.

File: src/io/flutter/samples/FlutterSampleManager.java [hunk 2]
```diff
@@ -37,6 +37,10 @@
     final List<FlutterSample> samples = new ArrayList<>();
     try {
       // TODO(pq): replace w/ index read from repo (https://github.com/flutter/flutter/pull/25515).
+
+      //https://docs.flutter.io/snippets/index.json
```
Comment (jacob314) [inline src/io/flutter/samples/FlutterSampleManager.java]: which of these locations is the right one to use?

File: src/io/flutter/samples/FlutterSample.java (hunk 1 above)
Comment (pq) [reply src/io/flutter/samples/FlutterSample.java]: Agreed.  Context from the description helps but yeah, these aren't great.  VERY open to ideas, though we'll likely want to stay consistent w/ what we're doing for VSCode.

/fyi @DanTup

File: src/io/flutter/samples/FlutterSampleManager.java (hunk 2 above)
Comment (pq) [reply src/io/flutter/samples/FlutterSampleManager.java]: @DanTup : is this now reliably https://master-docs.flutter.io/snippets/index.json?

**Comment by DanTup** [comment]
@pq Seems like I can't reply inline (maybe because it was merged), but here's my comments....

> labels

Here's how they currently show in VS Code:

screenshot 2019-01-15 at 2 57 16 pm


> is this now reliably https://master-docs.flutter.io/snippets/index.json?

There are two hosts:

- docs.flutter.io - this is tied to the stable version of Flutter and generally what we want to use - however, the Index file won't appear there until the next stable Flutter release (since the doc site is updated as part of the release)
- master-docs.flutter.io - this is a mirror of the site running from master. It's mainly for our own use (testing/previewing things). It's good for testing this, but less good to ship to users (at least, unless you know they're on the master branch - else they may see samples that then fail to create)

In VS Code, we currently have an empty default for the hostname, which results in this feature not showing. I have my local settings override this with the master-docs host which means it shows for me (for testing). When Flutter next has a stable release, I plan to update the default to docs.flutter.io and push a release (this means this feature generally won't be visible to users until the next stable release).

There was a little more discussion about this in https://github.com/Dart-Code/Dart-Code/pull/1391. If we're keen to enable the feature before a stable release, we may need to do some hacks (like just including a local registry of the snippets that are in the current live docs to use in the meantime).

**Comment by pq** [comment]
> If we're keen to enable the feature before a stable release, we may need to do some hacks (like just including a local registry of the snippets that are in the current live docs to use in the meantime).

FWIW I'm doing just this in the IntelliJ plugin.  👍 

My thinking has been to first check the URL (docs.flutter.io I guess) and fall back to a local copy when we ship but I'm open to discussing.  (You certainly shouldn't feel compelled to do the same unless you think it's a good idea.)

As for how it presents in VSCode, I think what you've got looks great!

Out of curiosity, I'm curious if you've considered stripping out the dartdoc markup from the description text?  (I'm thinking about that over on #2976.)

Cheers and thanks for all the thoughtful follow-ups.

**Comment by DanTup** [comment]
> My thinking has been to first check the URL (docs.flutter.io I guess) and fall back to a local copy

Sounds good to me - if you're planning on this, I'll do the same :-)

> Out of curiosity, I'm curious if you've considered stripping out the dartdoc markup from the description text? 

Yeah! I only noticed it when I posted the screenshot here so went to look up if we could use Markdown in those dropdowns.. we can't, do I think stripping the [brackets] off will make sense (I've also sorted it since that screenshot since it looked a bit weird in that order).

**Pull request merged**